### PR TITLE
Tiny refactor of method_missing definition

### PR DESCRIPTION
**Pull request by splattael**
* Removed unused var `args_node`
* ~~Handle `named_args_nodes` similar to `args_nodes`~~
* Drop `try` for non-nil `named_args`

### CODE REVIEW

**Comment by splattael** [comment]
Refs 8295d8ccae7a393d0490e648fcbf964625aab93b

File: src/compiler/crystal/semantic/method_missing.cr [hunk 1]
```diff
@@ -38,7 +38,7 @@ module Crystal
     def define_method_from_method_missing(method_missing, signature, original_call)
       name_node = StringLiteral.new(signature.name)
       args_nodes = [] of ASTNode
-      named_args_nodes = nil
+      named_args_nodes = [] of NamedArgument
```
Comment (asterite) [inline src/compiler/crystal/semantic/method_missing.cr]: But this will allocate an array all the time, even if not used...

Comment (splattael) [inline src/compiler/crystal/semantic/method_missing.cr]: @asterite Oh, that's true. I wanted to stay "symmetrical" with `args_nodes`.
I wonder, is allocating an array a hotspot here?

Anyway, I could revert the change for `named_args_nodes`. How about the other changes? Should I close this PR?

Comment (splattael) [inline src/compiler/crystal/semantic/method_missing.cr]: @asterite I've reverted to  `named_args_nodes = nil`. The other changes remained.

:eyes: ?

**Comment by splattael** [comment]
Thanks for merging! :purple_heart:
